app/models/job.rb

```diff
@@ -225,25 +225,14 @@ def is_active?
 
   def self.delete_older(ts, condition)
     _log.info("Queuing deletion of jobs older than: #{ts}")
-    cond = condition.blank? ? [] : [[condition].flatten.first, [condition].flatten[1..-1]]
-    cond[0] ||= []
-    cond[1] ||= []
-
-    ts_clause = "updated_on < ?"
-    cond[0].empty? ? cond[0] << ts_clause : cond[0] = "(#{cond[0]}) AND #{ts_clause}"
-    cond[1] << ts.utc
-
-    _log.info("cond.flatten: #{cond.flatten.inspect}")
-    ids = where(cond.flatten).pluck(:id)
-
+    ids = where("updated_on < ?", ts.utc).where(condition).pluck("id")
```

Comment:
Thanks. I could have sworn AR wasn't smart enough in the non sql column case. Verified you are right.